src/main/java/graphql/execution/ExecutionStepInfo.java

```diff
@@ -36,30 +30,37 @@
     private final Field field;
     private final GraphQLFieldDefinition fieldDefinition;
     private final ExecutionPath path;
-    private final boolean typeIsNonNull;
     private final Map<String, Object> arguments;
     private final ExecutionStepInfo parent;
 
-    private ExecutionStepInfo(GraphQLType type, GraphQLFieldDefinition fieldDefinition, Field field, ExecutionPath path, ExecutionStepInfo parent, boolean nonNull, Map<String, Object> arguments) {
+    private ExecutionStepInfo(GraphQLType type, GraphQLFieldDefinition fieldDefinition, Field field, ExecutionPath path, ExecutionStepInfo parent, Map<String, Object> arguments) {
         this.fieldDefinition = fieldDefinition;
         this.field = field;
         this.path = path;
         this.parent = parent;
-        this.type = type;
-        this.typeIsNonNull = nonNull;
+        this.type = assertNotNull(type, "you must provide a graphql type");
         this.arguments = arguments;
-        assertNotNull(this.type, "you must provide a graphql type");
+
     }
 
     /**
-     * This returns the type which is unwrapped if it was {@link GraphQLNonNull} wrapped
+     * This returns the type for the current step.
      *
      * @return the graphql type in question
      */
     public GraphQLType getType() {
         return type;
     }
 
+    /**
+     * This returns the type which is unwrapped if it was {@link GraphQLNonNull} wrapped
+     *
+     * @return the graphql type in question
+     */
+    public GraphQLType getUnwrapNonNullType() {
```

Comment:
Rather then have a convenience method here - why not just make the callers user GraphQLTypeUtil.unwrapNonNull() say.

If you want to keep this method then I would call it `getUnwrappedNonNullType`

Reply:
Trade off: it was used so much that it makes sense to add it here imho. But always a judgment call of course